### DISCUSSION TITLE
IntegretionTest와 RestDocTest에 코드가 겹치는 문제에 대하여

### DISCUSSION THREAD

**Discussion by lcqff** [BDD-CLUB/01-doo-re-back#59] [논의 사항]
현재 프로젝트는 Spring MVC환경에서 특정 HTTP 메서드 테스트를 하기 위해 `call-HTTP메서드명-Api`라는 커스텀 메서드를 사용하고 있습니다. 
해당 커스텀 메서드는 컨트롤러 테스트와 api 문서 테스트에서 사용되는데, 컨트롤러 테스트와 api 문서 테스트가 각각 상속받는 부모가 IntegretionTest와 RestDocsTest로 다르기 때문에 동일한 `call-HTTP메서드명-Api` 코드를 두 부모 클래스 모두 작성해야하는 문제가 있습니다. 

1. RestDocsTest가 IntegrationTest를 상속받게 하기
2. `call-HTTP메서드명-Api` 메서드를 모아두는 부모 클래스를 새로 만들어 IntegrationTest와 RestDocsTest가 상속받게 만들기

기타 의견이 있으면 작성 부탁드립니다~~!

**Comment by JJimini** [comment]
2번 좋습니다~~

**Comment by lcqff** [comment]
2번으로

**Comment by 02ggang9** [comment]
저도 2번 좋습니다~

**Comment by 02ggang9** [comment]
Discussion의 2번 방법을 사용하도록 결정되었습니다~

**Comment by shkisme** [comment]
굿입니다~ 클래스 이름은 ApiTestHelper로 하면 좋을 것 같네요~
